src/main/java/com/github/karsaig/approvalcrest/matcher/GsonProvider.java

```diff
@@ -286,4 +304,26 @@ public JsonElement serialize(Optional<T> src, Type typeOfSrc, JsonSerializationC
             return result;
         }
     }
+
+	private static class FixedZoneDateAdapter extends TypeAdapter<Date> {
+		static final TypeAdapterFactory FACTORY = new TypeAdapterFactory() {
+			@Override public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> typeToken) {
+				return Date.class.isAssignableFrom(typeToken.getRawType()) ? (TypeAdapter<T>) new FixedZoneDateAdapter() : null;
+			}
+		};
+
+		@Override public void write(JsonWriter jsonWriter, Date date) throws IOException {
+			if (date == null) {
+				jsonWriter.nullValue();
+			} else {
+				synchronized (FIXED_ZONE_DATE_FORMAT) {
```

Comment:
i would give each factory instance a FIXED_ZONE_DATE_FORMAT instance so they dont need to synchromize. That way the timezone could be a factory parameter in the future.

Reply:
@karsaig is a single instance of a TypeAdapter used only by a single thread, please?